### PR TITLE
feat: add clarity-lsp support

### PR DESCRIPTION
**Pull request by josecanhelp**
Screen Shot 2022-03-03 at 10 27 49 PM

Adds support for [clarity-lsp](https://github.com/hirosystems/clarity-lsp)

### CODE REVIEW

**Comment by josecanhelp** [comment]
I noticed that style-lint caused one of the workflow checks to fail. I ran it locally and it throws a ton of errors on the current codebase. However, I made the change that the linter requested in the workflow, rebased, and force pushed as the contribution docs requested.

File: lua/lspconfig/server_configurations/clarity_lsp.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    filetypes = { 'clar', 'clarity' },
+    root_dir = util.root_pattern '.git',
+  },
+  docs = {
+    description = [[
+`clarity-lsp` is a language server for the Clarity language. Clarity is a decidable smart contract language that optimizes for predictability and security. Smart contracts allow developers to encode essential business logic on a blockchain.
+
+To learn how to configure the clarity language server, see the [clarity-lsp documentation](https://github.com/hirosystems/clarity-lsp).
+
+```lua
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/clarity_lsp.lua:20]: We don't usually put this line, it's a duplicate of what is automatically generated.

Comment (josecanhelp) [reply lua/lspconfig/server_configurations/clarity_lsp.lua]: Got it. I removed that line. Thanks!

**Comment by josecanhelp** [comment]
I'm sure you're aware of this, but the style lint error in the github actions workflow is caused by a file I did not touch with this PR.

**Comment by mjlbach** [comment]
Yep, thanks!

**Pull request merged**